### PR TITLE
Issue 46 relay env

### PR DESCRIPTION
**Pull request by edipox**

### CODE REVIEW

File: apps/webapp/web/static/js/index.js [hunk 1]
```diff
@@ -34,17 +34,24 @@ function requireAuth(nextState, replace) {
   }
 }
 
+class Application extends React.Component{
+  render(){
+    const env = Auth.getEnvironment();
+    return (<Router history={browserHistory}
```
Comment (hisapy) [inline apps/webapp/web/static/js/index.js]: Please write it like:

``` javascript
return(
  <Router history="..."
    .....
  </Router>
);
```

File: apps/webapp/web/static/js/index.js [hunk 2]
```diff
@@ -34,17 +34,24 @@ function requireAuth(nextState, replace) {
...
+      render={applyRouterMiddleware(useRelay)}
+      environment={env}>
```
Comment (hisapy) [inline apps/webapp/web/static/js/index.js]: I think `environment={Auth.getEnvironment()}` would be better ... I don't know why suddenly I completely missed where `env` came from.

**Review by hisapy — CHANGES_REQUESTED** [review]
Just those minors changes and of course some "manual testing" to make sure that the environment's store is cleared up upon logout.

File: apps/webapp/web/static/js/index.js [hunk 3]
```diff
@@ -34,17 +34,41 @@ function requireAuth(nextState, replace) {
...
+    Auth.onLogout = ()=> this.handleLogout();
+  }
+
+  handleLogout(){
+    this.setState({ environment: Auth.getEnvironment() })
+  }
+
+  render(){
+    return (
+      <Router history={browserHistory} routes={routes}
```
Comment (hisapy) [inline apps/webapp/web/static/js/index.js]: What about ?

``` javascript
<Router history={browserHistory}
  render={applyRouterMiddleware(useRelay)}
  environment={this.state.environment}>
  {routes}
</Router>
```

File: apps/webapp/web/static/js/index.js [hunk 4]
```diff
@@ -34,17 +34,41 @@ function requireAuth(nextState, replace) {
...
+    <Route path="/admin" component={AdminLayout} onEnter={requireAuth} >
       <IndexRoute component={Hello}/>
       <Route path="star-wars" component={StarWarsApp} queries={StarWarsQueries}/>
       <Route path="graphiql" component={GraphiQL} />
     </Route>
-  </Router>,
+  </Route>
+)
+
+class Application extends React.Component{
```
Comment (hisapy) [inline apps/webapp/web/static/js/index.js:54]: If you want to make this file a little bit more _cleaner_ then you might want to move this `class` to its own file **app.js**. 

Anyway, I don't mind having this class here.

File: apps/webapp/web/static/js/index.js [hunk 5]
```diff
@@ -34,17 +34,41 @@ function requireAuth(nextState, replace) {
...
-  <Router history={browserHistory}
-    render={applyRouterMiddleware(useRelay)}
-    environment={() => Auth.getEnvironment()}>
-    <Route path="/" component={Hello}/>
+const routes = (
+  <Route path="/" >
+    <IndexRoute component={Hello}/>
     <Route path="/login" component={Login}/>
-    <Route path="/admin" component={AdminLayout} onEnter={requireAuth}>
+    <Route path="/admin" component={AdminLayout} onEnter={requireAuth} >
```
Comment (hisapy) [inline apps/webapp/web/static/js/index.js]: Can you please remove the whitespace before the `>`?

**Review by hisapy — CHANGES_REQUESTED** [review]
Looking good. Just a few touches to get this done.

File: apps/webapp/web/static/js/index.js (hunk 4 above)
Comment (edipox) [reply apps/webapp/web/static/js/index.js:54]: For now as the code is very simple I won't move this component. But if we get more and more routes and the Application component gets more complex, maybe we could separate the route logic in a Router and the Application component in another file.

In that case the Router could look something like this:

``` javascript
class Filter {

  static requireAuth(nextState, replace) {
    if (!Auth.loggedIn()) {
      replace({
        pathname: "/login",
        state: {
          nextPathname: nextState.location.pathname
        }
      })
    }
  }

  static verifySession(nextState, replace) {
    if (Auth.loggedIn()) 
      replace({pathname: "/admin"})
  }
}

export default  (
  <Route path="/">
    <IndexRoute component={Hello}/>
    <Route path="/login" component={Login} onEnter={Filter.verifySession}/>
    <Route path="/admin" component={AdminLayout} onEnter={Filter.requireAuth}>
      <IndexRoute component={Hello}/>
      <Route path="star-wars" component={StarWarsApp} queries={StarWarsQueries}/>
      <Route path="graphiql" component={GraphiQL}/>
    </Route>
  </Route>
)
```

Comment (hisapy) [reply apps/webapp/web/static/js/index.js:54]: 👍

**Review by hisapy — APPROVED** [review]
Looks nice and ready to go

**Pull request closed without merging**